.github/workflows/gen-check.yaml

```
@@ -1,22 +1,22 @@
 name: Clean Generation
 on:
+  push:
+    branches:
+      - main
   pull_request:
-env:
-  # Default minimum version of Go to support.
-  DEFAULT_GO_VERSION: 1.17
 jobs:
   generate-and-check:
     runs-on: ubuntu-latest
     steps:
       - name: Install Go
         uses: actions/setup-go@v3
         with:
-          go-version: ${{ env.DEFAULT_GO_VERSION }}
+          go-version: "1.20"
```

```suggestion
          go-version: "~1.20.2"
```

We should set a floor given known issues with versions <1.20.2.

---

SGTM